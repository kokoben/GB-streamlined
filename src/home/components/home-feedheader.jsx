import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchHomeSearchVideos, setHomeVideo } from '../actions/index';
import HomeSearchResultsSelector from '../selectors/current-home-search-results';

export const HomeFeedHeader = props => (
  <FeedHeader
    category=""
    placeholder="Search for videos..."
    fetchVideos={props.fetchHomeSearchVideos}
    results={props.homeSearchResults}
    currentResults={props.currentHomeSearchResults}
    setVideo={props.setHomeVideo}
  />
);

/* eslint-disable react/forbid-prop-types */
HomeFeedHeader.propTypes = {
  homeSearchResults: PropTypes.array.isRequired,
  currentHomeSearchResults: PropTypes.array.isRequired,
  fetchHomeSearchVideos: PropTypes.func.isRequired,
  setHomeVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  homeSearchResults: state.home.homeSearchResults,
  currentHomeSearchResults: HomeSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchHomeSearchVideos,
    setHomeVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(HomeFeedHeader);
