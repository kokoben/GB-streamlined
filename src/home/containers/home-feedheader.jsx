import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import FeedHeader from '../../components/feed-header';
import { fetchHomeSearchVideos, setHomeVideo } from '../actions/index';
import HomeSearchResultsSelector from '../selectors/current-home-search-results';

class HomeFeedHeader extends Component {
  render() {
    return (
      <FeedHeader
        category=""
        placeholder="Search for videos..."
        fetchVideos={this.props.fetchHomeSearchVideos}
        currentResults={this.props.currentHomeSearchResults}
        setVideo={this.props.setHomeVideo}
      />
    );
  }
}

/* eslint-disable react/forbid-prop-types */
HomeFeedHeader.propTypes = {
  fetchHomeSearchVideos: PropTypes.func.isRequired,
  currentHomeSearchResults: PropTypes.array.isRequired,
  setHomeVideo: PropTypes.func.isRequired,
};
/* eslint-enable */

const mapStateToProps = state => ({
  currentHomeSearchResults: HomeSearchResultsSelector(state),
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    fetchHomeSearchVideos,
    setHomeVideo,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(HomeFeedHeader);
