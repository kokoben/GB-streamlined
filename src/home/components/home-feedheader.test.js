import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HomeFeedHeader } from './home-feedheader';

describe('home-feedheader tests', () => {
  it('renders correctly', () => {
    const mockFetchHomeSearchVideos = jest.fn();
    const mockSetHomeVideo = jest.fn();

    const wrapper = shallow(<HomeFeedHeader
      homeSearchResults={[]}
      currentHomeSearchResults={[]}
      fetchHomeSearchVideos={mockFetchHomeSearchVideos}
      setHomeVideo={mockSetHomeVideo}
    />);

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
