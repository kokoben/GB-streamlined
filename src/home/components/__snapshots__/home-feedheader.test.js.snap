// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-feedheader tests renders correctly 1`] = `
<FeedHeader
  category=""
  currentResults={Array []}
  fetchVideos={[Function]}
  placeholder="Search for videos..."
  results={Array []}
  setVideo={[Function]}
/>
`;
